.github/workflows/build_tests.yml

```
@@ -0,0 +1,26 @@
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  Build Tests:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: set up JDK 21
+        uses: actions/setup-java@v3
```

actionlint (1.7.4)

the runner of "actions/setup-java@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)